traitsui/tests/editors/test_text_editor.py

```diff
@@ -34,16 +33,6 @@ class Foo(HasTraits):
     name = Str()
 
 
-@contextlib.contextmanager
-def launch_ui(gui_test_case, object, view):
-    ui = object.edit_traits(view=view)
-    try:
-        yield ui
-    finally:
-        with gui_test_case.delete_widget(ui.control):
```

Comment:
This `delete_widget` has a misleading name to me... it does not explicitly delete the widget. It runs the event loop upon exiting the context, until the widget `destroyed` signal is fired, see code [here](https://github.com/enthought/pyface/blob/351ec343843004abee3832bb79d1a4cac38df573/pyface/ui/qt4/util/event_loop_helper.py#L139-L169). So really this function is for processing all events after `ui.dispose()` is called, until the widget is destroyed. It is very much the same objective of calling `process_cascade_events` after `ui.dispose()`.

Comparing wrapping `ui.dispose` with `delete_widget` and calling `process_cascade_events` before and after `ui.dispose()`:
Differences:
(1) `delete_widget` does not let events to be processed before `ui.dispose` is called. All events, included ones that have been posted before `ui.dispose` is called, will be processed after `ui.dispose` is called.
(2) `delete_widget` will timeout if the widget is not destroyed within a time limit (default to one second)
(3) [I think; haven't tested] Since `delete_widget` will stop the event loop when the widget's destroyed signal is called, if there are other slots added to the destroyed signal and if those slots should push new events (cascade events again), then those events would not be processed, whereas `process_cascade_events` would carry on processing them.
(4) There is no wx implementation for `delete_widget`. A dependency on this single function results in tests not being run against wx.

(Side-note for (4): Given `GUI.process_events` does process all cascaded events in wx, an equivalence of `delete_widget` is not needed for wx in the test context.  Calling process_events would have been enough and we don't need to start the main event loop and then add a timer to stop it later when the widget is destroyed.)

Reply:
Thanks for a very detailed answer!